scripts/sipify.pl

```diff
@@ -456,6 +457,13 @@ sub fix_constants {
     return $line;
 }
 
+sub replace_macros {
+    my $line = $_[0];
+    $line =~ s/\bVAL_TRUE\b/``True``/g;
```

Comment:
What about the following, it would read a bit easier in the original header files.

```suggestion
    $line =~ s/\b\\c true\b/``True``/g;
```

Reply:
Do you mean instead of the macros?

Reply:
Yes, exactly. The proposal is to document C++ and derive the Python part (instead of documenting in a meta-language)

Reply:
I did think about this overnight, but I don't really like it. For a start it makes reviewing PRs and identifying missed raw true/false strings a bit harder, but more importantly, it would make it very difficult to find places missing the \c prefix (you'd need to use a negative lookbehind regex, which aren't widely supported). I'd also like to expand on this and add others like NULLPTR. I'm also slightly worried that \c isn't the correct annotation to use for doxygen, so i'd like it to be easy to revise this in future.

What about just an all caps TRUE/FALSE string instead?

Reply:
Good points.
`TRUE`, `FALSE` and `NULLPTR` sound readable.

Reply:
Sweet! Got any other suggestions?

I'm wondering if numeric constants should also be treated specially, e.g. "returns 0 when ..."

Reply:
"Returns ``0`` when" should work fine for C++ and Python. And it's probably worth considering upgrading return values to an enum in such cases.

Anyway, true, false and nullptr are probably the most important ones. Other things to look into would be more cross-linking inside the docs and adding code samples, so nothing that can be done as easy as the ones here.